src/github.com/matrix-org/dendrite/federationapi/routing/routing.go

```diff
@@ -165,7 +165,7 @@ func Setup(
 			roomID := vars["roomID"]
 			userID := vars["userID"]
 			return MakeJoin(
-				httpReq, request, cfg, query, roomID, userID,
+				httpReq.Context(), httpReq, request, cfg, query, roomID, userID,
```

Comment:
I feel like we should simplify this to just take `httpReq`...

Reply:
true. Rooted out some more deadcode #571